index.js

```diff
@@ -40,10 +40,27 @@ const format = (origStr, ...args) => Array.from(args).reduce(
   (str, arg, i) => str.replace(new RegExp(`\\{${i}\\}`, 'g'), arg), origStr
 )
 
+// Find a supported locale for the translated language names. First try the
+// browser's locale. If it isn't supported, try without a region code if it is,
+// otherwise return 'en'. A support for a locale is determined by having a
+// translation for the 'auto' string, this means that a new locale must be
+// complete, all the language names must be translated.
+const getSupportedLocale = (languages) => {
+  const locale = ps.getLocalized('general.useragent.locale', 'en')
+  if (languages.auto[locale]) {
+    return locale
+  }
+  const i = locale.indexOf('-')
```

Comment:
As some improvement, you could try to remove the dialect first. Then remove the region code.
Here is [the format of the locale codes](https://wiki.mozilla.org/L10n:Locale_Codes).